notebooks/brian_long_smFISH.ipynb

```diff
@@ -0,0 +1,131 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Reproduce Brian Long's results with Starfish\n",
+    "\n",
+    "The `brian_long_smFISH.zip` file needed to follow along with this notebook can be downloaded [here]()\n",
+    "\n",
+    "This notebook walks through a work flow that reproduces the smFISH result for one field of view using the starfish package. \n",
+    "It assumes that you have unzipped `brian_long_smFISH.zip` in the same directory as this notebook. Thus, you should see:\n",
+    "\n",
+    "raw/\n",
+    "brian_long_smFISH.ipynb"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "from starfish.io import Stack\n",
+    "import numpy as np"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# package this up same as before\n",
+    "experiment_json = os.path.expanduser('./brian_long_smFISH/experiment.json')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "s = Stack()\n",
+    "s.read(experiment_json)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# brian's data is uint, but starfish loads it as floats. Cast it to int here, which does not cause any loss of precision. \n",
+    "s.image._data = s.image._data.astype(np.uint16)\n",
+    "istack = s.image"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "OrderedDict([(<Indices.HYB: 'hyb'>, 1),\n",
+       "             (<Indices.CH: 'ch'>, 3),\n",
+       "             (<Indices.Z: 'z'>, 34),\n",
+       "             ('y', 2048),\n",
+       "             ('x', 2048)])"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# this doesn't work, do we want this abstraction? \n",
+    "# s.image.dtype"
```

Comment:
actually, what is a s.image? i feel as though image is probably not the right word for whatever it is because i think it's a tensor?

Reply:
`s.image` is the `ImageStack` (which I think of as the top-level object in our package, akin to the `numpy.ndarray` or `pandas.DataFrame`).

Reply:
in keeping the code parallel with the language of the spec-- should this be s.hybridization_images? I also find it confusing that there's a class called Stack that has an ImageStack in it ... any good ideas for other names?

Reply:
Keep in mind that Stack will go away in the long term, as operations should be happening on images, with the recipe gluing operations together.

Reply:
Wouldn't that mean that the Stack will change form from an array to an interface, but some kind of abstraction here will stay in the code base?

Reply:
Not really.  Check out https://github.com/spacetx/starfish/blob/268d43a22d0041aa2769963d7ba298ac77595e4e/tests/test_iss_data.py#L75.  This is the model for how things will be done in the future, where nothing really references experiment.json (which is what `Stack` represents) itself.  All commands should reference the data files directly.

Reply:
I don't get it. Let's whiteboard next week! That'll be super fun!